pkg/status/render.go

```diff
@@ -55,6 +55,17 @@ func FormatStatus(data []byte) (string, error) {
 	snmpTrapsStats := stats["snmpTrapsStats"]
 	title := fmt.Sprintf("Agent (v%s)", stats["version"])
 	stats["title"] = title
+	// return pared down version of status if cluster checks runner
+	if config.IsCLCRunner() {
+		renderStatusTemplate(b, "/header.tmpl", stats)
+		renderChecksStats(b, runnerStats, pyLoaderStats, pythonInit, autoConfigStats, checkSchedulerStats, inventoriesStats, "")
+		renderStatusTemplate(b, "/aggregator.tmpl", aggregatorStats)
+		renderStatusTemplate(b, "/endpoints.tmpl", endpointsInfos)
+		renderStatusTemplate(b, "/clusteragent.tmpl", dcaStats)
+		renderAutodiscoveryStats(b, stats["adEnabledFeatures"], stats["adConfigErrors"], stats["filterErrors"])
+
+		return b.String(), nil
+	}
```

Comment:
Is there a way to accomplish this without copy/pasting the desired function calls?  Could this be table-driven with some kind of boolean in each table row as to whether to include it for CLC?

My concern is that there's no information here to guide a future developer as to where to add a new template.  Maybe that engineer is working on the CLC runner and only adds it to this conditional, when it would be useful generally.  Or more likely, maybe that engineer is not familiar with CLC and just adds that template to the "non-CLC" case, not realizing that it would also be useful in the CLC case.

Reply:
@celenechang helped me come up with a better way to accomplish this using lambda functions. Let me know if the changes help to address your concerns.